integrationtests/controller/bundledeployment/status_test.go

```diff
@@ -0,0 +1,113 @@
+
+	"github.com/rancher/fleet/integrationtests/utils"
+	"github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = Describe("BundleDeployment Status Fields", func() {
+
+	BeforeEach(func() {
+		var err error
+		namespace, err = utils.NewNamespaceName()
+		Expect(err).ToNot(HaveOccurred())
+
+		ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}
+		Expect(k8sClient.Create(ctx, ns)).ToNot(HaveOccurred())
+
+		DeferCleanup(func() {
+			Expect(k8sClient.Delete(ctx, ns)).ToNot(HaveOccurred())
+		})
+	})
+
+	When("BundleDeployment changes", func() {
+		BeforeEach(func() {
+			cluster, err := createCluster("cluster", namespace, nil, namespace)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cluster).To(Not(BeNil()))
+			targets := []v1alpha1.BundleTarget{
+				{
+					BundleDeploymentOptions: v1alpha1.BundleDeploymentOptions{
+						TargetNamespace: "targetNs",
+					},
+					Name:        "cluster",
+					ClusterName: "cluster",
+				},
+			}
+			bundle, err := createBundle("name", namespace, targets, targets)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(bundle).To(Not(BeNil()))
+		})
+
+		AfterEach(func() {
+			Expect(k8sClient.Delete(ctx, &v1alpha1.Bundle{ObjectMeta: metav1.ObjectMeta{
+				Name:      "name",
+				Namespace: namespace,
+			}})).NotTo(HaveOccurred())
+
+		})
+
+		It("updates the status fields", func() {
+			bundle := &v1alpha1.Bundle{}
+			err := k8sClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: "name"}, bundle)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(bundle).To(Not(BeNil()))
+			Expect(bundle.Status.Display.State).To(Equal(""))
+
+			bd := &v1alpha1.BundleDeployment{}
+			Eventually(func() bool {
+				err = k8sClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: "name"}, bd)
+				if err != nil {
+					return false
+				}
+				Expect(err).NotTo(HaveOccurred())
+				return true
+			}).Should(BeTrue())
+
+			bd.Status.AppliedDeploymentID = bd.Spec.DeploymentID
+			bd.Status.Ready = true
+			bd.Status.NonModified = true
+			err = k8sClient.Status().Update(ctx, bd)
+			Expect(err).NotTo(HaveOccurred())
+			bd.Spec.StagedDeploymentID = bd.Spec.DeploymentID
+			err = k8sClient.Update(ctx, bd)
+			Eventually(func() bool {
+				err = k8sClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: "name"}, bd)
+				return err == nil && bd.Status.Display.State == "Ready" && bd.Spec.StagedDeploymentID == bd.Spec.DeploymentID
+			}).Should(BeTrue())
+
+			By("Updating the bundledeployment spec, updates status fields")
+			Expect(bd.Spec.Options.DeleteNamespace).ToNot(BeTrue())
+
+			Eventually(func() error {
+				err = k8sClient.Get(ctx, types.NamespacedName{Namespace: namespace, Name: "name"}, bd)
+				Expect(err).NotTo(HaveOccurred())
+				bd.Status.NonModified = false
+				return k8sClient.Status().Update(ctx, bd)
+			}).ShouldNot(HaveOccurred())
```

Comment:
Yes, this `Eventually` block is necessary because there are multiple updates. Prior to this block, there is another update on the bundle deployment spec.